test/repositories/project/project_local_source_test.dart

```diff
@@ -0,0 +1,111 @@
+import 'dart:math';
+
+import 'package:collection/collection.dart';
+import 'package:flutter_template/core/model/db/repository_db_entity.dart';
+import 'package:flutter_template/core/source/common/app_database.dart';
+import 'package:flutter_template/core/source/project_local_source.dart';
+import 'package:flutter_test/flutter_test.dart';
+
+void main() {
+  late AppDatabase database;
+  late ProjectLocalSource projectDao;
+
+  setUp(() async {
+    database = await $FloorAppDatabase.inMemoryDatabaseBuilder().build();
+    projectDao = database.projectLocalSource;
+  });
+
+  tearDown(() async {
+    await database.close();
+  });
+
+  group('Test projects db', () {
+    test('Get  projects from empty db, should return an empty list', () async {
+      final projects = await projectDao.getProjects().first;
+      expect(projects, []);
+    });
+    test('Insert one project to empty db, should return 1 project', () async {
+      final project = ProjectDbEntity(
+        id: 1,
+        name: 'Test projects',
+        description: 'Test project description',
+        url: 'test.com',
+        imageUrl: '',
+        language: 'ES',
+      );
+      await projectDao.insertProjects([project]);
+      final projects = await projectDao.getProjects().first;
+      expect(projects.length, 1);
```

Comment:
I tried that, but ProjectDbEntity doesn't have an equal operator overridden, since it doesn't check the properties, only the hash value both lists could have the same Project, but it can't detect that they are the same 
<img width="1179" alt="image" src="https://github.com/xmartlabs/flutter-template/assets/50491531/a7ec6c49-28f1-49af-a4d7-8e98a55c2bf4">